### ISSUE TITLE
Creating series with multiindex from dict with tuple keys with None

### ISSUE THREAD

**Issue by fumitoh** [pandas-dev/pandas#18505]
#### Code Sample

```python
import pandas as pd

data1={(1, 2): 3,
       (4, None): 6}

s1 = pd.Series(data=data1)

data2={(1, 2): 3,
       (None, 5): 6}

s2 = pd.Series(data=data2)
```

#### Problem description

`s1` has MultiIndex while `s2` has Index of tuple just because the second element of the `data2` has `None` as the first element of the key tuple.

```python
>>> s1
1  2      3.0
4  NaN    NaN
dtype: float64
>>> s2
(1, 2)       3
(None, 5)    6
dtype: int64
```
Is this a bug? or is this how it's supposed to be?

FYI, both of `mi1` and `mi2` below yield MultiIndex as I expect.

```python
mi1 = pd.MultiIndex.from_tuples([(1, 2), (4, None)])
mi2 = pd.MultiIndex.from_tuples([(1, 2), (None, 5)])
```

**Comment by toobaz** [comment]
This comes from
```
In [3]: pd.Index(list(data2))
Out[3]: Index([(1, 2), (None, 5)], dtype='object')
```

(so won't be fixed by #18496 ).

**Comment by toobaz** [comment]
Related to #18484
